vars/handlePipelineStepErrors.groovy

```diff
@@ -59,14 +64,19 @@ void call(Map parameters = [:], body) {
     try {
         if (config.echoDetails)
             echo "--- Begin library step of: ${config.stepName} ---"
-
-        body()
-    } catch (AbortException ae) {
+        if (!config.failOnError && config.stepTimeouts && config.stepTimeouts[config.stepName]) {
```

Comment:
How is the `timeout` related to `failOnError`?

Reply:
If someone sets `failOnError` to `true` I would assume he strives for 100% confidence in the pipeline's results and that does not fit to imposing timeouts and continuing without a clear result - at least that would be my interpretation.

Reply:
I was thinking about this too ...

My thoughts were that as @nevskrem wrote, that if you want to fail your builds on errors you also will not accept timeouts and continuation.